modules/programs/neovim/default.nix

```diff
@@ -0,0 +1,96 @@
+{ mode, config, pkgs, lib, soxin, ... }:
+
+with lib;
+let
+  cfg = config.soxin.programs.neovim;
+in
+{
+  options = {
+    soxin.programs.neovim = soxin.lib.mkSoxinModule {
+      inherit config;
+      name = "neovim";
+      includeKeyboardLayout = true;
+      includeTheme = true;
+      extraOptions = {
+        extraConfig = mkOption {
+          type = types.lines;
+          default = "";
+          example = ''
+            set nocompatible
+            set nobackup
+          '';
+          description = ''
+            Custom vimrc lines.
+          '';
+        };
+
+        plugins = mkOption {
+          type = with types; listOf (either package soxin.lib.modules.neovim.pluginWithConfigModule);
+          default = [ ];
+          defaultText = "The plugins added by the theme";
+          example = literalExample ''
+            with pkgs.vimPlugins; [
+              yankring
+              vim-nix
+              { plugin = vim-startify;
+                config = "let g:startify_change_to_vcs_root = 0";
+              }
+            ]
+          '';
+          description = ''
+            List of vim plugins to install optionally associated with
+            configuration to be placed in init.vim.
+          '';
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    # add all plugins installed by themes
+    { soxin.programs.neovim.plugins = cfg.theme.plugins; }
+
+    (optionalAttrs (mode == "NixOS") {
+      # TODO: Add support for NeoVim on NixOS.
+      # programs.neovim = {
+      #   enable = true;
+      #
+      #   # make it a default editor
+      #   defaultEditor = true;
+      #
+      #   # Create a symlink of nvim to vi and vim
+      #   viAlias = true;
+      #   vimAlias = true;
+      #
+      #   # Add support for NodeJS, Python 2 and 3 as well as Ruby
+      #   # withNodeJs = true;
+      #   # withPython3 = true;
+      #   withRuby = true;
+      #
+      #   # Add the Python's neovim plugin
+      #   # extraPython3Packages = ps: with ps; [ pynvim ];
+      #
+      #   configure = {
+      #     customRC = extraRC;
+      #     packages.myVimPackage = { inherit start opt; };
+      #   };
+      # };
+    })
+
+    (optionalAttrs (mode == "home-manager") {
+      # TODO: I wish home-manager had a defaultEditor as well!
+      home.sessionVariables = { EDITOR = "nvim"; };
+
+      programs.neovim = {
+        inherit (cfg) enable extraConfig plugins;
+
+        # Add the Python's neovim plugin
+        extraPython3Packages = ps: with ps; [ pynvim ];
+      }
+      # Create aliases
+      // (genAttrs [ "viAlias" "vimAlias" "vimdiffAlias" ] (name: true))
+      # Add support for NodeJS, Python 2 and 3 as well as Ruby
+      // (genAttrs [ "withNodeJs" "withPython3" "withRuby" ] (name: true));
```

Comment:
Are we planning on changing this once RFC 3 is merged and implemented?

Reply:
I'm surprised to see that Go didn't make that list :wink:

Reply:
I had these in my Soxin and thought it probably applies to everyone. I can move it back to my Soxincfg if you'd like to have a discussion around this.